src/inmanta/export.py

```diff
@@ -189,8 +190,36 @@ def _load_resources(self, types: Dict[str, Entity]) -> None:
                             instance.location,
                         )
 
+        self._load_resource_sets(types, resource_mapping)
         Resource.convert_requires(resource_mapping, ignored_set)
 
+    def _load_resource_sets(self, types: Dict[str, Entity], resource_mapping: Dict["Instance", "Resource"]) -> None:
+        """
+        load the resource_sets in a dict with as keys resource_ids and as values the name of the resource_set
+        the resource belongs to.
+        This method should only be called at the end of self._load_resources()
```

Comment:
I would explain the reason.

```suggestion
        This method should only be called after all resources have been extracted from the model.
```